scipy/special/special/tools.h

```diff
@@ -4,6 +4,8 @@
 
 #include "config.h"
 #include "error.h"
+#include <utility>
+#include <cstddef>
```

Comment:
All includes from the standard library need to go in [config.h](https://github.com/scipy/scipy/blob/main/scipy/special/special/config.h) and we use conditional compilation to swap between CUDA versions or standard library versions based on platform. For anything you use, you need to make sure that it is available in `libcudacxx` or `thrust`.

Reply:
I use the headers for `std::pair` and `std::size_t`. Let me check those in `config.h`.

Reply:
@izaid and I have our own GPU workstations to use. Don't worry about this aspect for now. It should be straightforward to adjust things after you get everything working.